lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -0,0 +1,82 @@
+const stream = require('stream');
+
+class ListRecordStream extends stream.Transform {
+    /**
+     * @constructor
+     * @param {Logger} logger logger
+     */
+    constructor(logger) {
+        super({ objectMode: true });
+        this._logger = logger;
+    }
+
+    /**
+     * Maps change stream operation type to
+     * backbeat supported types
+     * @param {string} operationType mongo opetation type
+     * @returns {string} supported operation type
+     */
+    _getType(operationType) {
+        switch (operationType) {
+            case 'insert':
+            case 'update':
+            case 'replace':
+                return 'put';
+            case 'delete':
+                return 'delete';
+            default:
+                this._logger.warn('Got unsupported operation', {
+                    method: 'ListRecordStream._getType',
+                    operationType,
+                });
+                return 'none';
+        }
+    }
+
+    /**
+     * parses json strings
+     * @param {string} blob sringified json
+     * @returns {Object} parsed json
+     */
+    _parseJson(blob) {
+        try {
+            let parsed = JSON.parse(blob);
+            if (typeof parsed === 'string') {
```

Comment:
How do you get this condition? I could not make it happen.

Reply:
Kafka-Connect MongoDB Source Connector is the one that'll be pushing messages into the topic used by this logReader. And so it is the one that over-stringifies the messages

Reply:
when consuming from kafka this is what i get:
```
"{\"_id\": {\"_data\": {\"$binary\": \"gmKorHgAAAABRjxfaWQAPGEtYnVja2V0AABaEAR3m6BArY5JKYbPqzz2tpQPBA==\", \"$type\": \"00\"}}, \"operationType\": \"delete\", \"ns\": {\"db\": \"metadata\", \"coll\": \"__metastore\"}, \"documentKey\": {\"_id\": \"a-bucket\"}}"
```

Reply:
That was parsed just fine in one go.

Reply:
Ok, this is the last thing left for me, I think the code is fine ( although I would use a while loop over recursion but whatever )  my only concern is why we need it in the first place. I have not seen this sort of double stringified before. If anything if it is a specific structure just doing parse twice would make more sense as its a fixed requirement.

Reply:
@vrancurel ^

Reply:
@miniscruff I had a call with @vrancurel yesterday, and we found that the value we get from the kafka message is actually doubly stringified, Kafka-connect is the one doing that and as far as i know that can't be changed.

So we can either do as we do now and parse it twice, or we could switch to BSON format and configure Kafka-connect to output in that format, in which case we'll only need to parse once. 

The issue with BSON though is that it's going to make debugging a bit more complicated if we want to see what get's pushed into the topic.

Side Note: i removed the recursion from the code, so now we parse twice at most. In case of debugging if we want to push a message manually the code can handle single parsing as well.

Reply:
i changed that actually

Reply:
Ok I see the new code, one moment